stdlib/enum.pyi

```diff
@@ -221,12 +227,24 @@ class IntFlag(int, Flag):
     def __rxor__(self: Self, n: int | Self) -> Self: ...
 
 if sys.version_info >= (3, 11):
-    class StrEnum(str, Enum):
+    class StrEnum(str, ReprEnum):
         def __new__(cls: type[Self], value: str | Self) -> Self: ...
         _value_: str
         @property
         def value(self) -> str: ...
 
+    class EnumCheck(StrEnum):
+        CONTINUOUS: str
+        NAMED_FLAGS: str
+        UNIQUE: str
+    CONTINUOUS = EnumCheck.CONTINUOUS
+    NAMED_FLAGS = EnumCheck.NAMED_FLAGS
+    UNIQUE = EnumCheck.UNIQUE
+
+    class verify:
+        def __init__(self, *checks: EnumCheck) -> None: ...
+        def __call__(self, enumeration: _S) -> _S: ...
```

Comment:
This TypeVar is bound to Type[Enum], would be good to give it a more descriptive name.